code/datums/components/temporary_mute.dm

```diff
@@ -0,0 +1,50 @@
+/datum/component/temporary_mute
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// A message to tell the user when they attempt to speak, if any
+	var/on_speak_message = ""
+	/// A message to tell the user when they become no longer mute, if any
+	var/on_unmute_message = ""
+	/// How long after the component's initialization it should be deleted. -1 means it will never delete
+	var/time_until_unmute = 5 MINUTES
+
+/datum/component/temporary_mute/Initialize(on_speak_message = "", on_unmute_message = "", time_until_unmute = 5 MINUTES)
+	. = ..()
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.on_speak_message = on_speak_message
+	src.on_unmute_message = on_unmute_message
+	src.time_until_unmute = time_until_unmute
+	if(time_until_unmute != -1)
+		QDEL_IN(src, time_until_unmute)
+
+
+/datum/component/temporary_mute/RegisterWithParent()
+	..()
+	RegisterSignal(parent, COMSIG_LIVING_SPEAK, PROC_REF(on_speak))
+	RegisterSignal(parent, COMSIG_XENO_TRY_HIVEMIND_TALK, PROC_REF(on_speak))
+
+/datum/component/temporary_mute/UnregisterFromParent()
+	..()
+	if(parent)
+		to_chat(parent, SPAN_NOTICE(on_unmute_message))
+		UnregisterSignal(parent, COMSIG_LIVING_SPEAK)
+		UnregisterFromParent(parent, COMSIG_XENO_TRY_HIVEMIND_TALK)
```

Comment:
This causes an infinite loop but otherwise looks good.
![Code_2024-01-14_12-59-32](https://github.com/cmss13-devs/cmss13/assets/25027759/e6584f89-924d-4e45-9634-fb9fc104320d)